Changed the structure of the release note slightly and fixed a few mi… (#124)
ballerina-platform/ballerina-release: release-notes/swan-lake-alpha3-release-note.md

layout: ballerina-blank-page
title: Release Note
---

> <em>This is the third alpha in a series of planned alphas and betas leading up to the Ballerina Swan Lake GA release.</em> 

Reviewer: ```suggestion
<em>This is the third alpha in a series of planned alphas and betas leading up to the Ballerina Swan Lake GA release.</em> 
```
I don't think we need to emphasize this sentence in italic using the `em` tag. The current wording also sounds fine as it doesn't emphasize too much on the milestones and gives the idea and meaning straight away.

@MaryamZi - WDYT?

Reviewer 2: +1 from me to emphasize the version a bit since these aren't yet the final release. But shouldn't we say

```suggestion
<em>This is the third alpha release in a series of planned alpha and beta releases leading up to the Ballerina Swan Lake GA release.</em> 
```

Author: It's just a style choice :). I was going for a casual voice. Also note the repetition of the word 'release', which was what I was trying to avoid. But since the rest of the release note is pretty formal, we can skip this and revisit the template for the next release for better user engagement with a semi-formal tone.

Reviewer: Changed as follows: "This Ballerina Swan Lake <VERSION> introduces new language features planned for the Swan Lake GA release. Moreover, this release includes improvements and bug fixes done to the compiler, runtime, standard library, and developer tooling after the Swan Lake <PREVIOUS_VERSION> release."

> Ballerina Swan Lake Alpha3 contains new language features, improvements, and bug fixes to the compiler, runtime, standard library, and developer tooling. 

Reviewer: ```suggestion
Ballerina Swan Lake Alpha3 contains new language features, improvements, and bug fixes to the compiler, runtime, standard library, and developer tooling. 
```
We specifically wanted to say that these alpha releases include the features that are planned for the GA release.

@MaryamZi - WDYT?

Reviewer 2: Yeah, it would be better if we can continue to say that this release includes the features planned for GA. But I think we can improve the current version too. How about something along the lines of

```suggestion
Ballerina Swan Lake Alpha3 introduces new language features planned for the Ballerina Swan Lake GA release. Moreover, this release includes improvements and bug fixes to the compiler, runtime, standard library, and developer tooling.
```

Reviewer 2: introduces vs includes because I think what we want to say here is that new features have been introduced between alpha2 and alpha3.

Author: +1. This was changed if we were going to mention that snippet at the beginning about alpha3 leading up to the Swan Lake GA release. BTW, what other components does Ballerina mainly consist of other than compiler, runtime, standard library and tooling? Something that's not mentioned here or didn't include any bug fixes or improvements?

Reviewer: Changed as follows:

"This Ballerina Swan Lake <VERSION> introduces new language features planned for the Swan Lake GA release. Moreover, this release includes improvements and bug fixes done to the compiler, runtime, standard library, and developer tooling after the after the Swan Lake <PREVIOUS_VERSION> release."

> ### Updating Ballerina to Swan Lake Alpha3

Reviewer: ```suggestion
### Updating Ballerina to Swan Lake Alpha3
```
Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/2050.

If you are using an **Update Tool version below 0.8.14**, execute the `ballerina update` command to update it. Next, execute the command below to update to Swan Lake Alpha3.

> `bal dist pull slalpha3`
<br><br>

> ### Installing Ballerina

Reviewer: ```suggestion
### Installing Ballerina
```
Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/2050.

Reviewer: @keizer619 - Please review the new changes in https://github.com/ballerina-platform/ballerina-dev-website/pull/2050.

> If you haven't installed Ballerina, download the [installers](/downloads/#swanlake) to install. <br><br>

Reviewer: ```suggestion
If you haven't installed Ballerina, download the [installers](/downloads/#swanlake) to install. <br><br>
```
We do not use shortened forms of words with the apostrophe in technical documentation.

> ### New in Ballerina Swan Lake Alpha3

Reviewer: ```suggestion
### New in Ballerina Swan Lake Alpha3
```
Better to say "What is new" here. Just saying "New" does not sound complete.

- [Language](#language)
- [Runtime](#runtime)
- [Standard Library](#standard-library)
- [Code to Cloud](#code-to-cloud)
- [Developer Tools](#developer-tools)
- [Breaking Changes](#breaking-changes) <br><br>

#### Language

##### Support for Module-level Variables with List, Mapping, and Error Binding Patterns

> Variable declarations with list, mapping, and error binding patterns are now allowed at the module level. Unlike simple variables, these variables must be initialized in the declaration.

Reviewer: ```suggestion
Variable declarations with list, mapping, and error binding patterns are now allowed at the module level. Unlike simple variables, these variables must be initialized in the declaration.
```
Fixed already.

var response = myclient->get("/some/endpoint", accHeaders);
```

> - Introduced header map and media type as optional arguments for entity-body client remote methods (POST, PUT, PATCH, DELETE, EXECUTE).

Reviewer: ```suggestion
- Introduced header map and media type as optional arguments for entity-body client remote methods (POST, PUT, PATCH, DELETE, EXECUTE).
```
Already fixed.

- Enabled read/listen for multiple emails in a single TCP connection.
Each POP3 or IMAP client/listener creation initiates the connection.
Then, the email sending, receiving, or listening operations can be performed many times.

> Finally, the client/listener has to be closed.

Reviewer: ```suggestion
Finally, the client/listener has to be closed.
```
Already fixed.

- The Ballerina HTTP listener can be configured to authenticate and authorize the inbound requests with a Basic Auth file user store.

> - Improved client and listener `SecureSocket` APIs of HTTP, GRPC, WebSocket, GraphQL, WebSub, WebSubHub, TCP, Email, NATS, STAN, and RabbitMQ modules.

Reviewer: ```suggestion
- Improved client and listener `SecureSocket` APIs of HTTP, GRPC, WebSocket, GraphQL, WebSub, WebSubHub, TCP, Email, NATS, STAN, and RabbitMQ modules.
```
Already fixed.

#### Introduced Local Repository Support

> - Apart from the Ballerina Central remote repository, you can now push packages to the local repository which can be found at `<user-home>/.ballerina/repositories/local`. Refer to the section on changes to CLI commands for information regarding pushing to the local repository.

Reviewer: ```suggestion
- Apart from the Ballerina Central remote repository, you can now push packages to the local repository which can be found at `<user-home>/.ballerina/repositories/local`. Refer to the section on changes to CLI commands for information regarding pushing to the local repository.
```
Already fixed.

  - Introduced pushing to the local repository. Passing `--repository=local` will push the Ballerina archive (.bala) to the local repository. For information about local repository support, see the [Ballerina Packages Changelist](<link>).
- Run `bal help <command>` to get more information on the command changes.

> - CLI Auto-Completion

Reviewer: ```suggestion
- CLI Auto-Completion
```
It is the verb form we have here and we do not need the hyphen.

Author: I believe auto-completion requires a hyphen. What do you mean verb form? Here's a usage example - https://npp-user-manual.org/docs/auto-completion/

Reviewer: For example, "log in" is the verb form in which we say "Log in to some app". "Log-in" with the hyphen is the noun/adjective form of it in which we say "The log-in will require a password". The same applies here.

Reviewer: +1
